web/src/navbar_alerts.js

```diff
@@ -70,6 +72,32 @@ export function should_show_server_upgrade_notification(ls) {
     return Date.now() > upgrade_nag_dismissal_duration;
 }
 
+export function maybe_show_empty_required_profile_fields_alert() {
+    const empty_required_profile_fields = realm.custom_profile_fields
+        .map((f) => ({
+            ...f,
+            value: people.my_custom_profile_data(f.id)?.value,
+        }))
+        .filter((f) => f.required && !f.value)
+        .map((f) => f.name);
+
+    $("#empty_required_profile_fields_warning").remove();
+    if (empty_required_profile_fields.length) {
+        let empty_required_profile_fields_string = util.format_array_as_list(
+            empty_required_profile_fields,
+            "long",
+            "conjunction",
+        );
+        empty_required_profile_fields_string += ". ";
```

Comment:
This sort of string manipulation is quite strange and probably not correct internationalization. Why are you doing it?

Reply:
I did that to add a period at the end of the list. Now I have added it in the "empty_profile_required_fields.hbs" itself:
```handlebars
{{ empty_required_profile_fields }}{{t '.'}}&nbsp;
```

Reply:
The current implementation adds a tooltip and '*' in front of the required profile fields but there is nothing to show which fields are empty explicitly. We can maybe change the font color of empty required profile fields or add a tip at the top of the edit profile form.

What do you think?

Reply:
OK, I have added a yellow border around empty required fields. You can take a look in the PR description.

Does this look good to you?

Reply:
I think red would be better. The color we use in the compose box when someone tries to send an empty message should be fine.

Reply:
OK, I have changed the border-color to red.